src/nginx/proto/status.proto

```diff
@@ -106,11 +106,22 @@ message ProcessStatus {
   repeated google.api_manager.proto.EspStatus esp_status = 6;
 }
 
+// Maps service configuration IDs to their corresponding traffic percentage.
+// Key is the service configuration ID, Value is the traffic percentage
+message ServiceConfigRollouts {
+  string service_name = 1;
+  string rollout_id = 2;
+  map<string, double> percentages = 3;
+}
+
 // Top-level endpoints status message
 message Status {
   // Overall server status
   ServerStatus server = 1;
 
   // Status for each process
   repeated ProcessStatus processes = 2;
+
+  // ESP rollouts
+  repeated ServiceConfigRollouts esp_rollouts = 3;
```

Comment:
Yes,  all string ids are fixed length,  we can define a max_config as 10.  so the size is fixed. 
In the timer,  only  write to shared memory when data is changed.  so to reduce chances of race condition (reading and writing in different process)